src/components/TreeView.js

```diff
@@ -56,9 +59,44 @@ function RowValue(props) {
     separator = ': ';
   }
 
+  if (position === 'TransactionEnvelope.signatures') {
+    checkSignatures({signatures: node, fetchedSigners});
+    return formatSignatureCheckState({node, separator})
+  }
+
+  if (position.match(/^TransactionEnvelope\.signatures\[[0-9]+\]\.signature$/)) {
+    return formatSignature({node, separator});
+  }
+
   return <span><strong>{node.type}</strong>{separator}{value}</span>
 }
 
+function checkSignatures({signatures, fetchedSigners}) {
+  // catch fetch signature errors
+  if (fetchedSigners === null || fetchedSigners === "PENDING") {
+    signatures.state = SIGNATURE.NOT_VERIFIED_YET;
+    return
+  } else if (fetchedSigners === "ERROR") {
+    signatures.state = SIGNATURE.INVALID;
+    return
+  }
+  
+  signatures.state = SIGNATURE.VALID;
+
+  for (var i = 0; i < signatures.nodes.length; i ++) {
+    const sig = signatures.nodes[i].nodes[1];
+    const signatureBuffer = Buffer.from(sig.value.value, 'base64');
```

Comment:
As a part of a separate PR we should probably move all the extra presentation functions here from `extrapolateFromXdr`. Right now we should probably leave the original value in a resulting object like:
```js
  if (object && object._isBuffer) {
    return {type: 'code', raw: object, value: new Buffer(object).toString('base64')};
  }
```
Without it we need to convert it back to bytes from base64 representation. Then we can do:
```js
const fetchedSignature = fetchedSigners.find(x => x.sig.equals(sig.raw));
```